python/mxnet/io.py

```diff
@@ -83,6 +85,85 @@ def getlabel(self):
         check_call(_LIB.MXDataIterGetLabel(self.handle, ctypes.byref(hdl)))
         return NDArray(hdl, False)
 
+class NumpyIter(DataIter):
+    """NumpyIter object in mxnet. Taking Numpy Array into dataiter. """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize with handle
+
+        Parameters
+        ----------
+        handle : DataIterHandle
+            the handle to the underlying C++ Data Iterator
+        """
+        super(NumpyIter, self).__init__(None)
```

Comment:
consider make DataIter an empty base class, and NumpyIter and MXDataIter inheritate from it